feat(very_good_core): add linux platform support

Crdzbird opened the pull request:
This PR Adds to the VG core the necessary files and configuration to support building and running the application on Linux.

Introduces a Linux application ID model, CMake build scripts, Flutter plugin registration, and application runner setup.

<!--
  Thanks for contributing!

  Provide a description of your changes below and a general summary in the title

  Please look at the following checklist to ensure that your PR can be accepted quickly:
-->

## Description

<!--- Describe your changes in detail -->

- Adds the necessary files and configuration to support building and running the application on Linux.
- Introduces a Linux application ID model, CMake build scripts, Flutter plugin registration, and application runner setup.


## Type of Change

<!--- Put an `x` in all the boxes that apply: -->

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [x] ✅ Build configuration change
- [x] 📝 Documentation
- [x] 🗑️ Chore
- [x] 🧪 Test

Crdzbird committed:
Adds Linux platform support

Adds the necessary files and configuration to support building
and running the application on Linux.

Introduces a Linux application ID model, CMake build scripts,
Flutter plugin registration, and application runner setup.

Crdzbird committed:
Generated plugins and updated naming on test

Crdzbird committed:
added pending test

marcossevilla reviewed (commented):
can we also remove the files related to `very_good_flame_game`? thanks! :)

Crdzbird committed:
removed generated_plugin from flame

Crdzbird committed:
Renames MyApplication to Application

Updates the application class name from MyApplication to Application
for consistency and clarity.

Refactors related files and code to reflect the new naming convention.

erickzanardo reviewed (commented):
Does this supports flavors? The reason that Linux wasn't in the very good core template was because that platform didn't supported flavors and adding it would make it inconsistent with the other platforms

marcossevilla commented:
closing this for now, we can reopen when there's a response
